plugins/rbac-backend/src/file-permissions/csv-file-watcher.ts

```diff
@@ -474,4 +478,19 @@ export class CSVFileWatcher extends AbstractFileWatcher<string[][]> {
     }
     this.csvFilePolicies.removedGroupPolicies = [];
   }
+
+  async cleanUpRolesAndPolicies(): Promise<void> {
+    const roleMetadatas =
+      await this.roleMetadataStorage.filterRoleMetadata('csv-file');
+    const fileRoles = roleMetadatas.map(meta => meta.roleEntityRef);
+
+    if (fileRoles.length > 0) {
+      this.csvFilePolicies.removedGroupPolicies =
+        await this.enforcer.getFilteredGroupingPolicy(1, ...fileRoles);
+      this.csvFilePolicies.removedPolicies =
+        await this.enforcer.getFilteredPolicy(0, ...fileRoles);
```

Comment:
```suggestion
      for (const fileRole of fileRoles) {
        this.csvFilePolicies.removedGroupPolicies.push(
          ...(await this.enforcer.getFilteredGroupingPolicy(1, fileRole)),
        );
        this.csvFilePolicies.removedPolicies.push(
          ...(await this.enforcer.getFilteredPolicy(0, fileRole)),
        );
      }
```
I ran into a bug whenever I disabled my CSV file. This resulting in me not having any roles or permissions removed. I couldn't make what you had originally had work for me, so I poked around and came up with a potential solution. Don't know if you want to incorporate it or if you want to still try to fix what you originally had.